src/lib.rs

```diff
@@ -291,13 +291,21 @@ fn get_log_file_path(
     if log_size > max_file_size {
       match rotation_strategy {
         RotationStrategy::KeepAll => {
-          fs::rename(
-            &path,
-            dir.as_ref().join(format!(
-              "{}.log",
-              chrono::Local::now().format("app-%Y-%m-%d")
-            )),
-          )?;
+          let to = dir.as_ref().join(format!(
+            "{}.log",
+            chrono::Local::now().format("app_%Y-%m-%d_%H-%M-%S")
+          ));
+          if to.is_file() {
+            // designated rotated log file name already exists
+            // highly unlikely but defensively handle anyway by adding .bak to filename
+            let mut to_bak = to.clone();
+            to_bak.set_file_name(format!(
+              "{}.bak",
```

Comment:
Is the `.bak` a common extensions for these sort of situations?

Reply:
to be honest I did not think long about it (https://en.wikipedia.org/wiki/Bak_file) and am open to other suggestions.
But keep in mind that this branch has a minor change of ever being called in any real world scenario so don't spend much time on it.

Reply:
Yeah you're right, didn't know this is actually a common extension